src/core/collection.ml

```diff
@@ -0,0 +1,16 @@
+
+module type VectorLike = sig 
```

Comment:
Can you pull this module (collection.ml) out of core and move it into containers. It's unlikely Imm.re will ever depend on this, it would be better to define core as the common shared subset, and containers vs. imm.re can be a user choice.